Use `rfc2822` date format

Changing `git-fast-import(1)` `--date-format` from `raw` to `rfc2822` to support more `GIT_COMMITTER_DATE` date values.

---

Rationale:

Using the [`rfc2822` _date format_ w/ git-fast-import](https://git-scm.com/docs/git-fast-import#_date_formats) improves compatibility with the [date formats supported by `GIT_COMMITER_DATE`](https://git-scm.com/docs/git-commit#_date_formats).

`GIT_COMMITER_DATE` was introduced to ghp-import in 8eeb52d (Respect GIT_COMMITTER environment variables (#81), 2020-07-24).

In contrast, the current in-use `raw` _date format_ supports _only_ the [_Git internal format_](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-Gitinternalformat) which is is very strict:

> Unlike the `rfc2822` format, this format is very strict. Any variation in formatting will cause fast-import to reject the value, and some sanity checks on the numeric values may also be performed. [ref](https://git-scm.com/docs/git-fast-import#Documentation/git-fast-import.txt-coderawcode)

Additionally despite the `raw`  format looks like it would map well with unix timestamps, when in use with git-fast-import it does not support all values, including the `0` (start of the epoch), as it requires a certain minimum value (for Git internal date parsing/detecting decisions): it must be after Jan 1, 2000 (git v0.99, [ref](https://github.com/git/git/commit/198b0fb635ed8a007bac0c16eab112c5e2c7995c#diff-1f93b2716aa69d2d974b91907e848ba6187ebef268c18eef8ba320c7c0aa03c5R242)) / Mar 3 1973 (git v1.5.2.2, [ref](https://github.com/git/git/commit/a1a5a6347b4ac1005c5d9c2b636e906bf6c8ec12#diff-1f93b2716aa69d2d974b91907e848ba6187ebef268c18eef8ba320c7c0aa03c5R421)).

A change to `rfc2822` therefore not only better integrates with the environment variable but also gives more options to users as test with different date formats for `GIT_COMMITTER_DATE` with ghp-import have shown.

The tests included all the [`GIT_COMMITTER_DATE` environment parameter](https://git-scm.com/docs/git-commit#_date_formats) supported three formats:

1. [Git internal format](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-Gitinternalformat)
2. [RFC 2822](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-RFC2822)
3. [ISO 8601](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt-ISO8601)

And perhaps some more (e.g. the unix timestamp as [`SOURCE_DATE_EPOCH`](https://reproducible-builds.org/docs/source-date-epoch/) suggest [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/mbed-cli/default.nix#L23) as an example of package building).

---

How did we test:

With git version 2.25.1 / Python 3.8.10 on Linux (Ubunut 20.04 LTS).

Test Setup:

~~~
$ mkdir -p build && git clone --mirror . build/mirror.git && git remote add origin build/mirror.git
$ python3 -m venv venv && . venv/bin/activate
(venv) $ pip install wheel
...
~~~

Example Test Run:

~~~
(venv) $ GIT_COMMITTER_DATE=100000000 make install lint docs
pip install -e .[dev]
...
 + a850261...29b40a6 gh-pages -> gh-pages (forced update)
~~~

Permutate on `GIT_COMMITTER_DATE` values as well as gh-pages w/ and w/o changes.

Failure Scenario w/ previous:

~~~
(venv) $ GIT_COMMITTER_DATE=100000000 make install lint docs
pip install -e .[dev]
...
python ./docs/build.py
ghp-import -p docs/ -b gh-pages -r origin -m "Update docs [skip ci]" -o
fatal: Invalid raw date "100000000" in ident: User <user@example.net> 100000000
fast-import: dumping crash report to .git/fast_import_crash_17782
Traceback (most recent call last):
  File "/home/user/project/ghp-import/venv/bin/ghp-import", line 11, in <module>
    load_entry_point('ghp-import', 'console_scripts', 'ghp-import')()
  File "/home/user/project/ghp-import/ghp_import.py", line 300, in main
    ghp_import(args.pop("directory"), **args)
  File "/home/user/project/ghp-import/ghp_import.py", line 279, in ghp_import
    run_import(git, srcdir, **opts)
  File "/home/user/project/ghp-import/ghp_import.py", line 196, in run_import
    sys.stdout.write(enc("Failed to process commit.\n"))
TypeError: write() argument must be str, not bytes
make: *** [Makefile:13: docs] Error 1
~~~

---

Context:

Recent changes in Mike [made me look into this argument](https://github.com/jimporter/mike/issues/99) and as the same code (to this regard) is in/from ghp-import, we reported here, too.

--- a/ghp_import.py
+++ b/ghp_import.py
@@ -176,7 +176,7 @@ def gitpath(fname):
 
 def run_import(git, srcdir, **opts):
     srcdir = dec(srcdir)
-    pipe = git.open('fast-import', '--date-format=raw', '--quiet',
+    pipe = git.open('fast-import', '--date-format=rfc2822', '--quiet',
                     stdin=sp.PIPE, stdout=None, stderr=None)
     start_commit(pipe, git, opts['branch'], opts['mesg'], opts['prefix'])
     for path, _, fnames in os.walk(srcdir, followlinks=opts['followlinks']):

